File: leetcode/cpp/PassedProblems/171-excel-sheet-column-number.cpp
/*
 * 171-excel-sheet-column-number.cpp
 *
 *  Created on: Aug 1, 2015
 *      Author: hanzhe
 */

#include <iostream>
using namespace std;

class Solution {
public:
    int titleToNumber(string s) {
        int len = s.length();
        int sum = 0, base = 1;
        for(int i = len-1; i >= 0 ; i --)
        {
        	sum += (s[i] - 'A' + 1) * base;
        	base *= 26;
        }
        return sum;
    }
};

File: leetcode/python/171-excel-sheet-column-number.py
#!/usr/bin/env python
# encoding: utf-8

"""

Related to question Excel Sheet Column Title

Given a column title as appear in an Excel sheet, return its corresponding column number.

For example:

    A -> 1
    B -> 2
    C -> 3
    ...
    Z -> 26
    AA -> 27
    AB -> 28
"""


# 一次过
class Solution(object):
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        if len(s) == 0: return 0
        return self.titleToNumber(s[0:-1]) * 26 + (ord(s[-1]) - ord('A') + 1)


